batch/batch/log_store.py

```diff
@@ -11,50 +11,76 @@
 
 
 class LogStore:
-    def __init__(self, batch_logs_bucket_name, instance_id, blocking_pool, *, project=None, credentials=None):
+    def __init__(self, batch_logs_bucket_name, instance_id, *, credentials=None):
         self.batch_logs_bucket_name = batch_logs_bucket_name
         self.instance_id = instance_id
         self.batch_logs_root = f'gs://{batch_logs_bucket_name}/batch/logs/{instance_id}/batch'
-        self.gcs = GCS(blocking_pool, project=project, credentials=credentials)
+
+        self.gcs_fs = GoogleStorageAsyncFS(credentials=credentials)
+        self.sem = asyncio.Semaphore(50)
 
         log.info(f'BATCH_LOGS_ROOT {self.batch_logs_root}')
         format_version = BatchFormatVersion(BATCH_FORMAT_VERSION)
         log.info(f'EXAMPLE BATCH_JOB_LOGS_PATH {self.log_path(format_version, 1, 1, "abc123", "main")}')
 
+    async def _write(self, url, data, binary=True):
```

Comment:
Hmm.

OK, so this makes me think two things:
1. The AsyncFS API is too narrow.
2. This function will reappear many time all across our code base when someone needs to read data from a filesystem.

I think we can improve the code base as a whole by pushing your new functions *into* AsyncFS. `pathlib` is the only example I can think of as inspiration for names. It has [`read_text`](https://docs.python.org/3/library/pathlib.html#pathlib.Path.read_text) and `write_text`. IMHO, I prefer explicit use of `encode` and `decode` in user code (it forces me to think about the encoding).

Concretely, my proposal is to add two operations to `AsyncFS`:
```python3
    @abc.abstractmethod
    async def open(self, url: str) -> ReadableStream:
        pass

    async def read(self, url: str) -> bytes:
        async with self.open(url) as reader:
            reader.read(data)
    # ...
    @abc.abstractmethod
    async def create(self, url: str, *, retry_writes: bool = True) -> WritableStream:
        pass

    async def write(self, url: str, data: bytes):
        async with self.create(url) as writer:
            writer.write(data)
```
And then to change `LogStore` to have, for example:
```python3
async def read_log_file(...):
    url = # ...
    return await self.fs.read(url).decode('utf-8')

async def write_log_file(..., data: bytes):
    url = # ...
    return await self.fs.write(url, data.encode('utf-8'))

# ...
async def read_spec_file(...):
    # ...
    await self.fs.read(url)
```